### PR TITLE
backblaze : delete operation is made asynchronously

### PR DESCRIPTION
**Pull request by Crashdown59**

### CODE REVIEW

File: oio/api/backblaze.py [hunk 1]
```diff
@@ -86,15 +86,20 @@ def _upload_chunks(self, conn, size, sha1, md5, temp):
...
                 temp.seek(0)
                 if try_number == 0:
+                    logger.debug('headers sent : %s'
+                                 % str(b2e.headers_send))
                     raise OioException('backblaze upload error: %s'
                                        % str(b2e))
+                else:
+                    val_tmp = pow(2, TRY_REQUEST_NUMBER - try_number)
+                    sleep = b2e.headers_received.get("Retry-After", val_tmp)
+                    time.sleep(sleep)
```
Comment (jkasarherou) [inline oio/api/backblaze.py]: please don't use time.sleep in all this API code, this will block everything
you should instead use `eventlet.sleep()`

Comment (fvennetier) [inline oio/api/backblaze.py]: And `val_tmp` is a really bad variable name.

File: oio/event/filters/content_cleaner.py [hunk 2]
```diff
@@ -1,26 +1,42 @@
...
 class ContentReaperFilter(Filter):
 
     def process(self, env, cb):
         event = Event(env)
+        config = ConfigParser.RawConfigParser(allow_no_value=True)
+        fd = open(self.conf.get('__file__'))
+        config.readfp(fd)
+        fd.close()
+        try:
+            application_key = config.get("filter:content_cleaner",
```
Comment (jkasarherou) [inline oio/event/filters/content_cleaner.py]: you don't do this!!

use `self.conf`

File: tools/oio-bootstrap.py [hunk 3]
```diff
@@ -582,6 +582,7 @@
 
 [filter:content_cleaner]
 use = egg:oio#content_cleaner
+backblaze_application_key = ${BACKBLAZE_APPLICATION_KEY}
```
Comment (jkasarherou) [inline tools/oio-bootstrap.py]: please no backblaze specific credentials in the filter config

File: oio/api/backblaze_http.py [hunk 4]
```diff
@@ -37,6 +38,47 @@ def _get_sha1(data):
...
+                       storage_method.bucket_name, storage_method.account_id)
+            raise BackblazeUtilsException(message)
+        key = '%s.%s' % (storage_method.account_id, storage_method.bucket_name)
+        if not renew:
+            authorization = BackblazeUtils.b2_authorization_list.get(key, None)
+            if authorization:
+                return authorization
+        config = ConfigParser.ConfigParser()
+        try:
+            config.readfp(open(application_key_path))
```
Comment (fvennetier) [inline oio/api/backblaze_http.py]: The file is never closed. You should use a _with_ statement or close is in a _finally_ block.

**Pull request merged**